src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -252,18 +253,23 @@ const StyledCheckbox = styled(
 
 export const Checkbox = forwardRef(
   (props: CheckboxProps, ref: React.RefObject<HTMLInputElement>) => {
-    const { id: propId, ...passProps } = props;
+    const { id: propId, status: propStatus, ...passProps } = props;
     return (
       <SuomifiThemeConsumer>
         {({ suomifiTheme }) => (
           <AutoId id={propId}>
             {(id) => (
-              <StyledCheckbox
-                theme={suomifiTheme}
-                id={id}
-                forwardedRef={ref}
-                {...passProps}
-              />
+              <CheckboxGroupConsumer>
+                {({ status: groupStatus }) => (
+                  <StyledCheckbox
+                    theme={suomifiTheme}
+                    id={id}
+                    forwardedRef={ref}
+                    status={!!propStatus ? propStatus : groupStatus}
+                    {...passProps}
+                  />
+                )}
+              </CheckboxGroupConsumer>
```

Comment:
Makes sense to use the React context pattern here I think. Same thing seems to be used in other places as well, like ExpanderGroup and RadioButtonGroup.